gulpfile.js

```diff
@@ -54,10 +42,10 @@ gulp.task('dist', function(done){
 });
 
 gulp.task('watch', function(){
-  (argv.q) ? gulp.watch(sources, ['typecheck', 'dist']) :
-  (argv.notest) ? gulp.watch(sources, ['lint', 'typecheck', 'dist']) :
-  gulp.watch(sources.concat(tests), ['test', 'lint', 'typecheck', 'dist']);
+  (argv.q) ? gulp.watch(sources, ['dist']) :
+  (argv.notest) ? gulp.watch(sources, ['lint', 'dist']) :
+  gulp.watch(sources.concat(tests), ['test', 'lint', 'dist']);
 });
 
 gulp.task('default', ['watch']);
-gulp.task('build', ['lint', 'test', 'typecheck', 'dist']);
\ No newline at end of file
+gulp.task('build', ['lint', 'test', 'dist']);
```

Comment:
As gulp-flowtype is broken and it is really expansive to typecheck on every build, it's a better idea to remove it from our gulp tests and run flow with a different strategy, in parallel.